sign/impl.go

```diff
@@ -64,7 +67,25 @@ func (*defaultImpl) VerifyFileInternal(signer *Signer, path string) (*SignedObje
 
 func (*defaultImpl) VerifyImageInternal(ctx context.Context, publickeyPath string, images []string) (*SignedObject, error) {
 	v := verify.VerifyCommand{KeyRef: publickeyPath}
-	return &SignedObject{}, v.Exec(ctx, images)
+	// We only pass one image at time for now
+	ref, err := name.ParseReference(images[0])
+	if err != nil {
+		return &SignedObject{}, v.Exec(ctx, images)
+	}
+
+	dig, err := crane.Digest(ref.String())
+	if err != nil {
+		return &SignedObject{}, v.Exec(ctx, images)
+	}
+
+	sigParsed := strings.ReplaceAll(dig, "sha256:", "sha256-")
+	obj := SignedObject{
+		digest:    dig,
+		reference: ref.String(),
+		sig:       fmt.Sprintf("%s:%s.sig", ref.Context().Name(), sigParsed),
+	}
+
+	return &obj, v.Exec(ctx, images)
```

Comment:
Let's move this code part out of the impl.go into sign.go. We already have a mocked version of `ParseReference` and we can add the same for `crane.Digest`.